azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -59,95 +64,152 @@ def __init__(self, wire_client, full_goal_state=False, base_incarnation=None):
         """
         uri = GOAL_STATE_URI.format(wire_client.get_endpoint())
         self.xml_text = wire_client.fetch_config(uri, wire_client.get_header())
-        xml_doc = parse_doc(self.xml_text)
+        self._xml_doc = parse_doc(self.xml_text)
+
+        self._ext_config_retriever = ext_config_retriever
+        self._wire_client = wire_client
+
+        self.ext_conf = None
+        self.hosting_env = None
+        self.shared_conf = None
+        self.certs = None
+        self.remote_access = None
+        self.artifacts_profile_blob_url = None
+        self.status_upload_blob_url = None
+        self.status_upload_blob_type = None
 
-        self.incarnation = findtext(xml_doc, "Incarnation")
-        self.expected_state = findtext(xml_doc, "ExpectedState")
-        role_instance = find(xml_doc, "RoleInstance")
+        # Retrieve goal state information
+        self.incarnation = findtext(self._xml_doc, "Incarnation")
+        role_instance = find(self._xml_doc, "RoleInstance")
         self.role_instance_id = findtext(role_instance, "InstanceId")
         role_config = find(role_instance, "Configuration")
         self.role_config_name = findtext(role_config, "ConfigName")
-        container = find(xml_doc, "Container")
+        container = find(self._xml_doc, "Container")
         self.container_id = findtext(container, "ContainerId")
-        lbprobe_ports = find(xml_doc, "LBProbePorts")
+        lbprobe_ports = find(self._xml_doc, "LBProbePorts")
         self.load_balancer_probe_port = findtext(lbprobe_ports, "Port")
-
         GoalState.ContainerID = self.container_id
 
-        fetch_full_goal_state = False
-        if full_goal_state:
-            fetch_full_goal_state = True
-            reason = 'force update'
-        elif base_incarnation is not None and self.incarnation != base_incarnation:
-            fetch_full_goal_state = True
-            reason = 'new incarnation'
-
-        if not fetch_full_goal_state:
-            self.hosting_env = None
-            self.shared_conf = None
-            self.certs = None
-            self.ext_conf = None
-            self.remote_access = None
-            return
+        if retrieval_mode != GoalStateRetrievalMode.Limited:
+            # Read properties necessary to retrieve the FastTrack goal state
+            self._retrieve_fabric_ext_conf_properties()
 
-        logger.info('Fetching new goal state [incarnation {0} ({1})]', self.incarnation, reason)
```

Comment:
can you point me to the equivalent of this message in the new code? It is essential for debugging

Reply:
We have two messages:
1) (mentioned below) when we've determined our goal state but we haven't processed any extensions
2) After we've processed the extensions:
Finish and save data from previous mode {0}. New mode is {1}
(slightly different one for startup)